src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -620,12 +623,12 @@ public static function setTrustedHeaderName($key, $value)
      *
      * @return string The header name
      *
-     * @throws \InvalidArgumentException
+     * @throws InvalidTrustedHeaderException
      */
     public static function getTrustedHeaderName($key)
     {
         if (!array_key_exists($key, self::$trustedHeaders)) {
-            throw new \InvalidArgumentException(sprintf('Unable to get the trusted header name for key "%s".', $key));
+            throw new InvalidTrustedHeaderException($key, null, sprintf('Unable to get the trusted header name for key "%s".', $key));
```

Comment:
Isn't an `InvalidArgumentException` localized in the component enough and more appropriate?